client/src/components/Atoms/ListAndDetail.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { ReactNode } from 'react'
+import styled from 'styled-components'
+import { Button, Card, Classes, Colors, InputGroup } from '@blueprintjs/core'
+
+/**
+ * A set of components to render a list-and-detail panel
+ *
+ * Example usage:
+ *
+ *  <ListAndDetail>
+ *    <List>
+ *      <ListItem>Item 1</ListItem>
+ *      <ListItem selected>Item 2</ListItem>
+ *    </List>
+ *    <Detail>
+ *      {detailsForSelectedListItem}
+ *    </Detail>
+ *  </ListAndDetail>
+ */
+
+export const ListAndDetail = styled(Card)`
+  display: grid;
+  grid-template-columns: 1fr 3fr;
+  overflow-y: auto;
+  padding: 0;
+  width: 100%;
+`
+
+// ---------- List ----------
+
+const ListContainer = styled.div`
+  border-right: 1px solid ${Colors.LIGHT_GRAY2};
+`
+
+const ListSearch = styled(InputGroup).attrs({
+  leftIcon: 'search',
+  type: 'search',
+})`
+  margin: 16px 12px;
+`
+
+const ListItems = styled.ul<{ withTopBorder?: boolean }>`
+  ${props =>
+    props.withTopBorder && `border-top: 1px solid ${Colors.LIGHT_GRAY2};`}
+  list-style: none;
+  margin: 0;
+  padding: 0;
+  padding-bottom: 40px;
+`
+
+interface IListProps {
+  search?: {
```

Comment:
Yes good q! I've been thinking of composable components vs. traditional props as a spectrum - you can take things as far as you'd like in either direction. Composable components provide more flexibility (you can use the provided composable component, e.g. `search={<ListSearch />}`, or your own, e.g. `search={<MyCustomSearch />}`), while traditional props can be simpler / more convenient (one less import, one less puzzle piece to put together).

In this case, I decided that there was little to be gained by making search composable. But am of course open to changing that down the road should the need arise for more customizable search elements.

My opinion isn't strong for this specific case, and I could convince myself of either direction

Reply:
Thanks for sharing your thinking. I agree there's not really any flexibility needed here.

As I've been playing with more "logic-less" components that just do layout/styling, I've been enjoying how composability lets the markup show the layout structure a bit more clearly. E.g. in this case:

```ts
<List>
  <ListSearch />
  <ListItems>
     <ListItem />
     <ListItem />
     <ListItem />
  </ListItems>
</List>
```
Here I can see that there's a search bar on top of the list items. Or I could easily omit it or move it to the bottom (though as mentioned, we don't need that here).

Reply:
Yes, great point in favor of composable components!

A side note: I've often encountered the need to create props for some of the composable elements, instead of specifying them all as children, e.g.

```
<List search={<ListSearch />}>
  <ListItems>
    <ListItem />
    <ListItem />
  </ListItem>
</List>
```

This can be helpful if, for example, you need to know whether a search element has been provided or not, and you need to toggle styles accordingly. Or if you have to render an element somewhere that isn't purely before or after the other elements